clouddriver-aws/src/main/groovy/com/netflix/spinnaker/clouddriver/aws/provider/agent/ReservationReportCachingAgent.groovy

```diff
@@ -178,15 +178,22 @@ class ReservationReportCachingAgent implements CachingAgent, CustomScheduledAgen
 
     ConcurrentHashMap<String, OverallReservationDetail> reservations = new ConcurrentHashMap<>()
     ConcurrentHashMap<String, Collection<String>> errorsByRegion = new ConcurrentHashMap<>()
-    Observable
-      .from(accounts.sort { it.name })
-      .flatMap({ credential ->
-        extractReservations(reservations, errorsByRegion, credential).subscribeOn(scheduler)
-    })
-      .observeOn(scheduler)
-      .toList()
-      .toBlocking()
-      .single()
+
+    Map<NetflixAmazonCredentials, Future> tasks = accounts.collectEntries { NetflixAmazonCredentials credential ->
+      [
+        (credential) : reservationReportPool.submit {
+          extractReservations(reservations, errorsByRegion, credential)
```

Comment:
my rx is bad enough that I can't tell whether the previous version would have used a thread per region in each credential or not but either way I don't think it matters.